### PR TITLE
added velocity loss

### PR DESCRIPTION
**Pull request by HaKai-2062**

### CODE REVIEW

**Review by Rex109 — COMMENTED** [review]
I think there's a big issue with this... seems like it's triggering even if my velocity is perfectly stable and/or 0. plus i need you to leave the default ini settings to their default value and just add the new settings, other than this stuff it's a super-duper idea to add to the velometer!

**Comment by HaKai-2062** [comment]
ahhh oki i can change values to default but changing to velocityLoss color on consistent/stable speed was intended and dont you think it should work like that? 🤔

**Comment by HaKai-2062** [comment]
I can make it not trigger on stable velocity if u want...
Just my opinion that velocity loss color on stable velocity would look cool

**Comment by Rex109** [comment]
> ahhh oki i can change values to default but changing to velocityLoss color on consistent/stable speed was intended and dont you think it should work like that? 🤔

Honestly i've never seen something like that, maybe i'm wrong but then it will beome a velocity gain color change instead of velocity loss

**Comment by HaKai-2062** [comment]
yea in codjumper atleast that's how it usually works. Stable velocity and 0 is velocity loss color
i can just rename velocity color to velocity gain color and then should be good ^^

File: iw3velometer/dllmain.cpp [hunk 1]
```diff
@@ -132,6 +151,7 @@ bool setConfig()
     ini.SetValue("Config", "guiKey", str);
 
     ini.SetValue("Config", "resetOnDeath", iw3velometerConfig.resetOnDeath ? "True" : "False");
+	ini.SetValue("Config", "velocityLossCheck", iw3velometerConfig.velocityLossCheck ? "True" : "False");
```
Comment (Rex109) [inline iw3velometer/dllmain.cpp]: Here

File: iw3velometer/dllmain.cpp [hunk 2]
```diff
@@ -208,6 +228,19 @@ HRESULT __stdcall hookedEndScene(IDirect3DDevice9* pDevice) {
 
     if (vel > maxvel)
         maxvel = vel;
+
```
Comment (Rex109) [inline iw3velometer/dllmain.cpp]: Here

File: iw3velometer/dllmain.cpp [hunk 3]
```diff
@@ -218,8 +251,10 @@ HRESULT __stdcall hookedEndScene(IDirect3DDevice9* pDevice) {
     str.str(std::string());
     str << vel;
 
-    if (showHud)
+    if (showHud && (accel ||(!accel && !iw3velometerConfig.velocityLossCheck)))
         font->DrawText(NULL, str.str().c_str(), -1, &veloRectangle, DT_NOCLIP | DT_CENTER | DT_BOTTOM, D3DCOLOR_ARGB((int)(255 * iw3velometerConfig.velocityColor[3]), (int)(255 * iw3velometerConfig.velocityColor[0]), (int)(255 * iw3velometerConfig.velocityColor[1]), (int)(255 * iw3velometerConfig.velocityColor[2])));
+	else if (showHud && !accel && iw3velometerConfig.velocityLossCheck)
```
Comment (Rex109) [inline iw3velometer/dllmain.cpp]: Here

File: iw3velometer/dllmain.cpp [hunk 4]
```diff
@@ -261,7 +296,9 @@ HRESULT __stdcall hookedEndScene(IDirect3DDevice9* pDevice) {
         ImGui::DragInt2("Max velocity position", iw3velometerConfig.maxVelocityPos);
         ImGui::DragInt2("Velocity position", iw3velometerConfig.velocityPos);
         ImGui::ColorEdit4("Max velocity color", iw3velometerConfig.maxVelocityColor);
-        ImGui::ColorEdit4("Velocity color", iw3velometerConfig.velocityColor);
+        ImGui::ColorEdit4("Velocity gain color", iw3velometerConfig.velocityColor);
+        ImGui::ColorEdit4("Velocity loss color", iw3velometerConfig.velocityLossColor);
+		ImGui::DragInt("Velocity loss delay", &iw3velometerConfig.loss_velocity);
```
Comment (Rex109) [inline iw3velometer/dllmain.cpp]: Here

File: iw3velometer/dllmain.cpp [hunk 5]
```diff
@@ -300,10 +337,10 @@ HRESULT __stdcall hookedEndScene(IDirect3DDevice9* pDevice) {
         ImGui::Text("Toggle gui key");
 
         ImGui::Checkbox("Reset on death", &iw3velometerConfig.resetOnDeath);
-
-        ImGui::NewLine();
-        ImGui::NewLine();
-
+		ImGui::Checkbox("Show velocity loss", &iw3velometerConfig.velocityLossCheck);
```
Comment (Rex109) [inline iw3velometer/dllmain.cpp]: Here

File: iw3velometer/dllmain.cpp [hunk 6]
```diff
@@ -46,20 +46,23 @@ struct iw3velometerConfig_s {
 
     bool showMaxVelocity = true;
     int fontSize = 60;
-
+	int loss_velocity = 30;
```
Comment (Rex109) [inline iw3velometer/dllmain.cpp]: Here

**Review by Rex109 — CHANGES_REQUESTED** [review]
Seems good now... the only thing i would ask you before merging is to fix the indention of some of the code, after that i'll give it a test and i'll let you know if i'll merge it or not.

**Review by Rex109 — APPROVED** [review]
Perfect! I'll merge the changes!

**Comment by Rex109** [comment]
And thanks for the font crash fix <3

**Pull request merged**